### ISSUE TITLE
Broken Heatsource Recipe

### ISSUE THREAD

**Issue by NielsPilgaard** [zeroneye/Powah#81]
#### Version: (make sure you are on the latest version before reporting):

- Minecraft: 1.15.2
- Forge: 1.15.2-31.1.75
- Powah: Powah-1.15.2-1.1.14

#### Issue description:
An error is printed in the log about a broken recipe

#### Steps to reproduce:
* Just a world
* Use /reload
* Read latest.log

https://gist.github.com/NillerMedDild/260f1c4d99cef1096de292416b8e8c74

**Comment by owmii** [comment]
Hey NillerMedDild!, I think the heat sources list in `general_common.toml` has some incorrect value.

**Comment by NielsPilgaard** [comment]
Oh woops, that's where the heatSource list is, thanks for the help :P

**Comment by NielsPilgaard** [comment]
So err, I replaced the heatSources config I had with `heatSources = []`, and reloaded, the log showed the same error.

**Comment by owmii** [comment]
Oh, ok i will see what's the problem.

**Comment by BlueAgent** [comment]
I tried with `heatSources = []` and then doing `/reload` with:

- jei-1.16.4-7.6.1.71
- Lollipop-1.16.4-3.2.6
- Powah-1.16.4-2.3.12

And it doesn't seem to print that error to the log anymore it seems. [Logs](https://gist.github.com/BlueAgent/18908a33af6db2fccb6f22ac3daa0150)

**Comment by NielsPilgaard** [comment]
I still see this error in Enigmatica 6 0.4.0 in-dev
Here's the latest log, both with a filled out heatSources array and empty, which gave the same result oddly enough: https://gist.github.com/NillerMedDild/28b54931b5f3cd716ec5656bdf971442

HeatSources config from first attempt, before reload:
`heatSources = ["minecraft:lava=1000", "minecraft:magma_block=1200", "byg:cryptic_magma_block=1200", "quark:brimstone=1200", "byg:brimstone=1200", "betterendforge:brimstone=1200", "botania:blaze_block=1400", "botania:blaze_lantern=1400", "minecraft:fire_coral=1500", "powah:blazing_crystal_block=1600", "powah:nitro_crystal_block=2200"]`

**Comment by BlueAgent** [comment]
In the log I can see stuff from some other mods too. However Powah seems to be massively spamming the log you posted. Unfortunately I can't easily load a whole modpack into development environment but will try do some static analysis to see if I can spot anything (as I can't reproduce with Powah and JEI alone as shown in my previous post).

<details>
<summary>Click for Examples</summary>

**Powah** (for comparison)
```py
...
[20:15:15] [Render thread/ERROR]: Found a broken recipe: owmii.powah.client.compat.jei.HeatSourceCategory$Recipe@7a7f8241
Outputs:
Inputs:
net.minecraft.item.ItemStack: [[1 air minecraft:air]]

java.lang.IllegalArgumentException: ItemStack value must not be empty. 1 air minecraft:air
	at mezz.jei.util.ErrorUtil.checkNotEmpty(ErrorUtil.java:184) ~[jei:7.6.1.71]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:87) ~[jei:7.6.1.71]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:28) ~[jei:7.6.1.71]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:94) ~[jei:7.6.1.71]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:75) ~[jei:7.6.1.71]
...
```
**Nature's Aura**
```py
...
[20:15:15] [Render thread/ERROR]: Found a broken recipe: Nature's Aura naturesaura:kubejs_generated_6156_minecraft_air class de.ellpeck.naturesaura.recipes.AnimalSpawnerRecipe
Outputs:
net.minecraft.item.ItemStack: [[1 air minecraft:air]]
Inputs:
net.minecraft.item.ItemStack: [[1 birth_spirit naturesaura:birth_spirit], [1 blitz_rod thermal:blitz_rod], [1 blitz_powder thermal:blitz_powder]]

java.lang.IllegalArgumentException: ItemStack value must not be empty. 1 air minecraft:air
	at mezz.jei.util.ErrorUtil.checkNotEmpty(ErrorUtil.java:184) ~[jei:7.6.1.71]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:87) ~[jei:7.6.1.71]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:28) ~[jei:7.6.1.71]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:94) ~[jei:7.6.1.71]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:75) ~[jei:7.6.1.71]
...
```
**MythicBotany**
```py
...
[20:15:15] [Render thread/ERROR]: Found a broken recipe: MythicBotany mythicbotany:petal_apothecary/wither_aconite class vazkii.botania.common.crafting.RecipePetals
Outputs:
net.minecraft.item.ItemStack: [[1 air minecraft:air]]
Inputs:
net.minecraft.item.ItemStack: [[1 air minecraft:air]]

java.lang.IllegalArgumentException: ItemStack value must not be empty. 1 air minecraft:air
	at mezz.jei.util.ErrorUtil.checkNotEmpty(ErrorUtil.java:184) ~[jei:7.6.1.71]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:87) ~[jei:7.6.1.71]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:28) ~[jei:7.6.1.71]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:94) ~[jei:7.6.1.71]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:75) ~[jei:7.6.1.71]
...
```
**Occultism**
```py
...
[20:15:15] [Render thread/ERROR]: Found a broken recipe: Occultism occultism:kubejs_generated_2896_minecraft_air class com.github.klikli_dev.occultism.crafting.recipe.CrushingRecipe
Outputs:
net.minecraft.item.ItemStack: [[2 air minecraft:air]]
Inputs:
net.minecraft.item.ItemStack: [[1 arcane_andesite_ore emendatusenigmatica:arcane_andesite_ore, 1 arcane_granite_ore emendatusenigmatica:arcane_granite_ore, 1 arcane_diorite_ore emendatusenigmatica:arcane_diorite_ore, 1 arcane_sand_ore emendatusenigmatica:arcane_sand_ore, 1 arcane_gravel_ore emendatusenigmatica:arcane_gravel_ore, 1 arcane_netherrack_ore emendatusenigmatica:arcane_netherrack_ore, 1 arcane_blackstone_ore emendatusenigmatica:arcane_blackstone_ore, 1 arcane_basalt_ore emendatusenigmatica:arcane_basalt_ore, 1 arcane_soul_soil_ore emendatusenigmatica:arcane_soul_soil_ore, 1 arcane_end_stone_ore emendatusenigmatica:arcane_end_stone_ore, 1 arcane_gabbro_ore emendatusenigmatica:arcane_gabbro_ore, 1 arcane_c_limestone_ore emendatusenigmatica:arcane_c_limestone_ore, 1 arcane_scoria_ore emendatusenigmatica:arcane_scoria_ore, 1 arcane_weathered_limestone_ore emendatusenigmatica:arcane_weathered_limestone_ore, 1 arcane_jasper_ore emendatusenigmatica:arcane_jasper_ore, 1 arcane_marble_ore emendatusenigmatica:arcane_marble_ore, 1 arcane_slate_ore emendatusenigmatica:arcane_slate_ore, 1 arcane_mossy_stone_ore emendatusenigmatica:arcane_mossy_stone_ore, 1 arcane_brimstone_ore emendatusenigmatica:arcane_brimstone_ore, 1 arcane_subzero_ash_ore emendatusenigmatica:arcane_subzero_ash_ore, 1 arcane_blue_netherrack_ore emendatusenigmatica:arcane_blue_netherrack_ore, 1 arcane_nylium_soul_soil_ore emendatusenigmatica:arcane_nylium_soul_soil_ore, 1 arcane_ether_stone_ore emendatusenigmatica:arcane_ether_stone_ore, 1 arcane_cryptic_stone_ore emendatusenigmatica:arcane_cryptic_stone_ore, 1 arcane_flavolite_ore emendatusenigmatica:arcane_flavolite_ore, 1 arcane_sulphuric_rock_ore emendatusenigmatica:arcane_sulphuric_rock_ore, 1 arcane_violecite_ore emendatusenigmatica:arcane_violecite_ore, 1 arcane_raw_marble_ore emendatusenigmatica:arcane_raw_marble_ore]]

java.lang.IllegalArgumentException: ItemStack value must not be empty. 2 air minecraft:air
	at mezz.jei.util.ErrorUtil.checkNotEmpty(ErrorUtil.java:184) ~[jei:7.6.1.71]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:87) ~[jei:7.6.1.71]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:28) ~[jei:7.6.1.71]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:94) ~[jei:7.6.1.71]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:75) ~[jei:7.6.1.71]
...
```
</details>

Edit: Added some debug code to the mod and compiled to then added it to Enigmatica 6 to detect what is going on and it seems like it's being filled with a bunch of air blocks somehow...
<details>
<summary>Debug Output</summary>

```py
...
[21Feb2021 10:14:20.865] [Render thread/DEBUG] [mezz.jei.load.PluginCaller/]: Registering recipes: jei:minecraft took 1.426 s
[21Feb2021 10:14:20.865] [Render thread/DEBUG] [mezz.jei.load.PluginCaller/]: Registering recipes: justenoughbeacons:justenoughbeacons ...
[21Feb2021 10:14:20.870] [Render thread/DEBUG] [mezz.jei.load.PluginCaller/]: Registering recipes: justenoughbeacons:justenoughbeacons took 5.707 ms
[21Feb2021 10:14:20.870] [Render thread/DEBUG] [mezz.jei.load.PluginCaller/]: Registering recipes: powah:main ...
[21Feb2021 10:14:20.900] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:magma_block} -> 1200}
[21Feb2021 10:14:20.901] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{quark:brimstone} -> 1200}
[21Feb2021 10:14:20.901] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:fire_coral} -> 1800}
[21Feb2021 10:14:20.902] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{powah:blazing_crystal_block} -> 2800}
[21Feb2021 10:14:20.902] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{powah:nitro_crystal_block} -> 8000}
[21Feb2021 10:14:20.902] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{botania:blaze_block} -> 1600}
[21Feb2021 10:14:20.906] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.906] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:lava}/net.minecraft.fluid.LavaFluid$Source@5cb79c20 -> 1000}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.907] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE INIT: HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:20.910] [Render thread/DEBUG] [mezz.jei.load.PluginCaller/]: Registering recipes: powah:main took 39.65 ms
[21Feb2021 10:14:20.910] [Render thread/DEBUG] [mezz.jei.load.PluginCaller/]: Registering recipes: tmechworks:jei ...
[21Feb2021 10:14:20.910] [Render thread/DEBUG] [mezz.jei.load.PluginCaller/]: Registering recipes: tmechworks:jei took 3.200 ?s
[21Feb2021 10:14:20.910] [Render thread/DEBUG] [mezz.jei.load.PluginCaller/]: Registering recipes: refinedstorage:plugin ...
[21Feb2021 10:14:20.910] [Render thread/DEBUG] [mezz.jei.load.PluginCaller/]: Registering recipes: refinedstorage:plugin took 1.600 ?s
...
```
Then later on:
```py
...
[21Feb2021 10:14:31.942] [Render thread/DEBUG] [mezz.jei.recipes.RecipeManager/]: Loading recipes: create:sandpaper_polishing
[21Feb2021 10:14:31.942] [Render thread/DEBUG] [mezz.jei.recipes.RecipeManager/]: Loading recipes: industrialforegoing:mycelial_meatallurgic
[21Feb2021 10:14:31.942] [Render thread/DEBUG] [mezz.jei.recipes.RecipeManager/]: Loading recipes: powah:heat.sources
[21Feb2021 10:14:31.942] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{minecraft:magma_block} -> 1200}
[21Feb2021 10:14:31.942] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{quark:brimstone} -> 1200}
[21Feb2021 10:14:31.942] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{minecraft:fire_coral} -> 1800}
[21Feb2021 10:14:31.942] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{powah:blazing_crystal_block} -> 2800}
[21Feb2021 10:14:31.942] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{powah:nitro_crystal_block} -> 8000}
[21Feb2021 10:14:31.942] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{botania:blaze_block} -> 1600}
[21Feb2021 10:14:31.942] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:31.942] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:31.943] [Render thread/ERROR] [mezz.jei.recipes.RecipeManager/]: Found a broken recipe: HeatSourceRecipe{Block{minecraft:air} -> 1600}
Outputs:
Inputs:
net.minecraft.item.ItemStack: [[1 air minecraft:air]]

java.lang.IllegalArgumentException: ItemStack value must not be empty. 1 air minecraft:air
	at mezz.jei.util.ErrorUtil.checkNotEmpty(ErrorUtil.java:184) ~[jei:7.6.1.65]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:87) ~[jei:7.6.1.65]
	at mezz.jei.plugins.vanilla.ingredients.item.ItemStackHelper.getUniqueId(ItemStackHelper.java:28) ~[jei:7.6.1.65]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:94) ~[jei:7.6.1.65]
	at mezz.jei.recipes.RecipeMap.addRecipe(RecipeMap.java:75) ~[jei:7.6.1.65]
	at mezz.jei.recipes.RecipeManager.addRecipe(RecipeManager.java:143) ~[jei:7.6.1.65]
	at mezz.jei.recipes.RecipeManager.addRecipeTyped(RecipeManager.java:105) ~[jei:7.6.1.65]
	at mezz.jei.recipes.RecipeManager.addRecipes(RecipeManager.java:98) ~[jei:7.6.1.65]
	at mezz.jei.recipes.RecipeManager.<init>(RecipeManager.java:90) ~[jei:7.6.1.65]
	at mezz.jei.load.PluginLoader.getRecipeManager(PluginLoader.java:142) ~[jei:7.6.1.65]
	at mezz.jei.startup.JeiStarter.start(JeiStarter.java:73) ~[jei:7.6.1.65]
	at mezz.jei.startup.ClientLifecycleHandler.onRecipesLoaded(ClientLifecycleHandler.java:128) ~[jei:7.6.1.65]
	at mezz.jei.startup.ClientLifecycleHandler.lambda$new$1(ClientLifecycleHandler.java:98) ~[jei:7.6.1.65]
	at net.minecraftforge.eventbus.EventBus.doCastFilter(EventBus.java:247) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.lambda$addListener$11(EventBus.java:239) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:302) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.eventbus.EventBus.post(EventBus.java:283) ~[eventbus-4.0.0.jar:?]
	at net.minecraftforge.client.ForgeHooksClient.onRecipesUpdated(ForgeHooksClient.java:699) ~[forge:?]
	at net.minecraft.client.network.play.ClientPlayNetHandler.func_199525_a(ClientPlayNetHandler.java:1314) ~[?:?]
	at net.minecraft.network.play.server.SUpdateRecipesPacket.func_148833_a(SourceFile:27) ~[?:?]
	at net.minecraft.network.play.server.SUpdateRecipesPacket.func_148833_a(SourceFile:15) ~[?:?]
	at net.minecraft.network.PacketThreadUtil.func_225383_a(SourceFile:21) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213166_h(SourceFile:144) ~[?:?]
	at net.minecraft.util.concurrent.RecursiveEventLoop.func_213166_h(SourceFile:23) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213168_p(SourceFile:118) ~[?:?]
	at net.minecraft.util.concurrent.ThreadTaskExecutor.func_213160_bf(SourceFile:103) ~[?:?]
	at net.minecraft.client.Minecraft.func_195542_b(Minecraft.java:947) ~[?:?]
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:607) ~[?:?]
	at net.minecraft.client.main.Main.main(Main.java:184) ~[1.16.5.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_271]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_271]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_271]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_271]
	at net.minecraftforge.fml.loading.FMLClientLaunchProvider.lambda$launchService$0(FMLClientLaunchProvider.java:51) ~[forge-1.16.5-36.0.10.jar:36.0]
	at cpw.mods.modlauncher.LaunchServiceHandlerDecorator.launch(LaunchServiceHandlerDecorator.java:37) [modlauncher-8.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:54) [modlauncher-8.0.9.jar:?]
	at cpw.mods.modlauncher.LaunchServiceHandler.launch(LaunchServiceHandler.java:72) [modlauncher-8.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.run(Launcher.java:82) [modlauncher-8.0.9.jar:?]
	at cpw.mods.modlauncher.Launcher.main(Launcher.java:66) [modlauncher-8.0.9.jar:?]
[21Feb2021 10:14:31.943] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{minecraft:air} -> 1600}
[21Feb2021 10:14:31.943] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{minecraft:air} -> 1600}
...
```
Then even later when browsing the recipes it only display the valid ones (which match what's in the config):
```py
...
[21Feb2021 10:14:55.567] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{minecraft:magma_block} -> 1200}
[21Feb2021 10:14:55.568] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{quark:brimstone} -> 1200}
[21Feb2021 10:14:55.568] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{minecraft:fire_coral} -> 1800}
[21Feb2021 10:14:55.568] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{powah:blazing_crystal_block} -> 2800}
[21Feb2021 10:14:57.255] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{powah:nitro_crystal_block} -> 8000}
[21Feb2021 10:14:57.256] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE getBlock(): HeatSourceRecipe{Block{botania:blaze_block} -> 1600}
...
```
</details>

So yeah I'm kinda confused how this is happening for now.
[Here are the full logs](https://gist.github.com/BlueAgent/64b9dec958d1f4312e1395b2771e0a05) with Enigmatica 6 with Lollipop updated and a custom build of Powah with debug info added.

**Comment by BlueAgent** [comment]
Ah, finally. I found what's causing it.
This is debug output:
```
[21Feb2021 12:29:29.861] [Render thread/INFO] [powah/]: HEAT SOURCE RECIPE ALL: [
Block{quark:brimstone} -> 1200,
Block{minecraft:fire_coral} -> 1800,
Block{botania:blaze_block} -> 1600,
Block{minecraft:air} -> 1600,
Block{minecraft:lava} -> 1000,
Block{powah:nitro_crystal_block} -> 8000,
Block{powah:blazing_crystal_block} -> 2800,
Block{minecraft:magma_block} -> 1200
]
```
This is the heat sources in Enigmatica 6 v0.3.13:
```
heatSources = [
    "minecraft:lava=1000",
    "minecraft:magma_block=1200",
    "minecraft:fire_coral=1800",
    "powah:nitro_crystal_block=8000",
    "powah:blazing_crystal_block=2800",
    "botania:blaze_block=1600",
    "botania:blaze_lantern=1600",
    "quark:brimstone=1200"
]
```

Turns out `botania:blaze_lantern` does not exist, I think it's intended to be `quark:blaze_lantern` instead. After changing that it no longer results in those air blocks being added and errors in the log.

However, it would be good for Powah to have some log output for invalid heat sources (air and non-existing blocks), and to not add them. Seems like there is already some checking but it needs additional checks. Will make a PR for this.

[Debug Output](https://gist.github.com/BlueAgent/1a7d0ef44d0244ea0481fab1ca3569e6)

[Debug Ouptut 2](https://gist.github.com/BlueAgent/e5376130f99061073226916824f84aee) after changing`botania:blaze_lantern` to `quark:blaze_lantern`.

Also the reason why the reload resulted in the same result after changing to empty is because Powah seems to only read the config on the initial load. Might be worth changing Powah to use the datapack system more.
